go-controller/pkg/ovn/endpoints.go

```diff
@@ -147,6 +147,17 @@ func (ovn *Controller) handleNodePortLB(node *kapi.Node) error {
 	if physicalIPs, _ = ovn.getGatewayPhysicalIPs(gatewayRouter); physicalIPs == nil {
 		return fmt.Errorf("gateway physical IP for node %q does not yet exist", node.Name)
 	}
+	// if new services controller run a full sync on all services
+	// services that have host network endpoints, are nodeport, external IP or ingress all have unique
+	// per-node load balancers. Since we cannot determine which services those are without significant parsing
+	// just sync all services
+	if ovn.svcController != nil {
+		if err := ovn.svcController.RequestFullSync(); err != nil {
```

Comment:
why can't we remove this handler directly and add a node handler to the new service controller?

Reply:
the thing is, the service controller can handle nodes updated directly, is just this codepath should be avoided if the servicecontroller runs, it keeps the abstraction 

> Since we cannot determine which services those are without significant parsing just sync all services

it is less expensive that triggering syncs for each service, all services are stored locally in the informer cache

Reply:
"the thing is, the service controller can handle nodes updated directly, is just this codepath should be avoided if the servicecontroller runs, it keeps the abstraction"

yes, it may be better off having the services controller also watch nodes, but that is a code optimization we can do later. I can make a github issue for it.

"it is less expensive that triggering syncs for each service, all services are stored locally in the informer cache"

Yeah, perhaps my comment wasn't great here. My point was really to figure out which services to sync, we would be doing a bunch of the same logic that the sync services is already doing (iterate through all services, find nodeport, external ip, ingress, find any service with host endpoint, find any service with external traffic policy local), so there is no point in duplicating the same logic to determine to add the key to the queue, just add them all and let the sync decide what to do. Realistically this isnt optimal right now because it will result in more ovn-nbctl commands because we dont have a local ovn cache yet, but in the future this would be less taxing.

Reply:
ahh ok, :+1: